CHANGELOG.md

```diff
@@ -185,9 +185,9 @@ poshRuntime.stopOfferService(myServiceDescription);
 poshRuntime.findService({"ServiceA", iox::capro::AnyInstanceString});
 
 // after
-discoveryInfo.offerService(myServiceDescription);
-discoveryInfo.stopOfferService(myServiceDescription);
-discoveryInfo.findService("ServiceA", Wildcard);
+serviceDiscovery.offerService(myServiceDescription);
```

Comment:
Totally right @MatthiasKillat. I already discussed this with Spitzi aka @FerdinandSpitzschnueffler. The possibility to offer a service without having a port was some kind of workaround to support iceoryx usage with ara::com. As there were no client/server ports back then, an iceoryx user needed a way to discover an ara::com service that only consists of methods (They had a custom comm channel for methods). So we needed this to create entries in our service registry without having related ports. As we have now client/server and want to have no ara::com specific things, we should remove `offerService()` and `stopOfferService()` here. And the `ServiceRegistry` and `ServiceDiscovery` should be less ara::com-ish but ara::com compatible. 

cc: @mossmaurice @elBoberido

Reply:
@mossmaurice @elBoberido @MatthiasKillat I would create separate PRs for removing the `offerService` and `stopOfferService` methods (since this will effect the service discovery tests) and for removing the application port (won't be used anymore). 
I also like to reduce the scope of this PR, i.e. only merge the new service discovery class. An example with complete readme will be part of another PR, probably after `startFindService` is adapted to search for events as well.
Is that okay with you?

Reply:
@budrus We should still think about how to properly use it with callbacks, compared to polling with the `findService` call. There are also efficiency considerations where we can use the counter to quickly detect nothing has changed etc. (this can be used for caching, I already discussed it with @FerdinandSpitzschnueffler), but also requires some thought.

Ideally we can use caching and the polling call combined with some condition variable/listener to obtain a reactive API with callbacks as well, but this is the next step. Some prototypying there led to some complications, but it is doable if time permits.